execution/executor-benchmark/src/fake_executor.rs

```diff
@@ -319,35 +317,48 @@ impl FakeExecutor {
     }
 }
 
-impl TransactionBlockExecutor<BenchmarkTransaction> for FakeExecutor {
+impl TransactionBlockExecutor<Transaction> for FakeExecutor {
     fn execute_transaction_block(
-        transactions: Vec<BenchmarkTransaction>,
+        transactions: Vec<Transaction>,
         state_view: CachedStateView,
     ) -> Result<ChunkOutput> {
         let transaction_outputs = FAKE_EXECUTOR_POOL.install(|| {
             transactions
                 .par_iter()
-                .map(|txn| match &txn.extra_info {
-                    Some(extra_info) => match &extra_info {
-                        ExtraInfo::TransferInfo(transfer_info) => {
-                            Self::handle_transfer(transfer_info, &state_view)
+                .map(|txn| match &txn {
+                    Transaction::UserTransaction(user_txn) => match user_txn.payload() {
+                        aptos_types::transaction::TransactionPayload::EntryFunction(f) => {
+                            match (
+                                *f.module().address(),
+                                f.module().name().as_str(),
+                                f.function().as_str(),
+                            ) {
+                                (AccountAddress::ONE, "coin", "transfer") => Self::handle_transfer(
+                                    user_txn.sender(),
+                                    bcs::from_bytes(&f.args()[0]).unwrap(),
+                                    bcs::from_bytes(&f.args()[1]).unwrap(),
+                                    &state_view,
+                                ),
+                                (AccountAddress::ONE, "aptos_account", "transfer") => {
+                                    Self::handle_account_creation(
+                                        user_txn.sender(),
+                                        bcs::from_bytes(&f.args()[0]).unwrap(),
+                                        bcs::from_bytes(&f.args()[1]).unwrap(),
+                                        &state_view,
+                                    )
+                                },
+                                _ => unimplemented!(),
+                            }
```

Comment:
goal of this is to replace transaction generator for executor benchmark with the transaction generator used for transaction emitter. And that will allow us to have executor benchmark and forge closer, and have richer benchmarks for executor benchmark.

having BenchmarkTransactions for increasing number of transaction types, and inside transaction emitter, doesn't seem very clean. Having duplicated transaction definitions for seems quite cumbersome. 
This way FakeExecutor doesn't need to support them all (and executor benchmark will), but for the one it does - having them handled by re-parsing /re-reading is closer to what is needed to be done by VM.

@grao1991 @msmouse let me know what do you think

Reply:
I intentionally introduced a new BenchmarkTransaction (instead of using existing https://github.com/aptos-labs/aptos-core/blob/main/execution/executor/src/mock_vm/mod.rs) because:
1. I want to skip the parsing.
2. Future extensibility and flexibility. I want the executor benchmark to have the capability to allow taking an extra field as the input to control the behavior of the fake executor, and/or doing whatever we want without having a real transaction.


For your purpose of unifying transaction generators for both benchmark and forge, how about writing a transformer from Transaction to BenchmarkTransaction?